### PR TITLE
Add support for Haiku lights

### PR DESCRIPTION
**Pull request by davidwkeith**
I was about to do this myself, but then I noticed @stabbylambda already did.

This refactor seems correct now that Big Ass Fans has multiple device types (and more in the works from what I understand.)

Thoughts?

### CODE REVIEW

File: src/BigAssApi.js [hunk 1]
```diff
@@ -0,0 +1,94 @@
+const dgram = require("dgram");
+const BigAssFan = require('./BigAssFan');
+const BigAssLight = require('./BigAssLight');
+const BigAssProperty = require('./BigAssProperty');
+const { syncingCallback, retryCall, myLogWrapper } = require('./utils');
+
+function FanMaster (numberOfExpectedDevices) {
+    this.allDevices = {}; // Dictionary of fan name -> BigAssDevice
+    this.sonnectionOpen = false;
```
Comment (sean9keenan) [inline src/BigAssApi.js:9]: Typo - this is supposed to be `connectionOpen` - typo made in refactoring

File: Examples/largerExample.js [hunk 2]
```diff
@@ -1,4 +1,4 @@
-var bigAssApi = require("../BigAssApi");
+var bigAssApi = require("../");
 bigAssApi.logging = false;
```
Comment (sean9keenan) [inline Examples/largerExample.js:2]: Removed in this PR

File: src/BigAssApi.js [hunk 3]
```diff
@@ -0,0 +1,94 @@
+const dgram = require("dgram");
+const BigAssFan = require('./BigAssFan');
+const BigAssLight = require('./BigAssLight');
+const BigAssProperty = require('./BigAssProperty');
+const { syncingCallback, retryCall, myLogWrapper } = require('./utils');
+
+function FanMaster (numberOfExpectedDevices) {
+    this.allDevices = {}; // Dictionary of fan name -> BigAssDevice
```
Comment (sean9keenan) [inline src/BigAssApi.js:8]: This change should be in README.md - and might impact Homekit support, so I'm not sure how I feel about the renaming purely from a backwards compatibility perspective

**Review by sean9keenan — CHANGES_REQUESTED** [review]
This PR looks good, a very much a needed refactoring on multiple levels!

A few concerns I have:
 - `README.md` hasn't been updated to reflect variables changed
 - Breaks compatibility with [Homebridge for Big ass fans](https://github.com/sean9keenan/homebridge-bigAssFans) because of some name changes I believe. And I was stupid there and didn't pin a version or scope it to a major version, so would have to release a new version that wouldn't grab this release.
 - A few minor issues commented on below.

I also don't currently have my fan installed as I moved semi-recently - so testing is a bit tricky for me right now.

In terms of making these changes I'm open to doing them myself - but if you want to make the proposed changes I'm more than open 😄 

Thanks @davidwkeith and @stabbylambda!

**Comment by davidwkeith** [comment]
Great feedback, will try to get all the changes added this week.

Thanks!

**Comment by stabbylambda** [comment]
Oh yeah, I never really got around to finishing up everything that I wanted to do here. I've got a fan and two standalone lights, but I haven't purchased the light module for the fan itself, so I'm not sure if that would break with these updates.

**Comment by superphilp** [comment]
Really looking forward to support for the Haiku Lights, any progress on getting this added to the API and Homebridge platform?

**Comment by davidwkeith** [comment]
I got tied up at work, will make some time to look into the rest of the changes needed to merge the fix by the end of the week.

File: src/BigAssApi.js (hunk 1 above)
Comment (davidwkeith) [reply src/BigAssApi.js:9]: :heavy_check_mark:

File: Examples/largerExample.js (hunk 2 above)
Comment (davidwkeith) [reply Examples/largerExample.js:2]: :heavy_check_mark:

**Pull request closed without merging**